File: j-LIB-core/src/main/java/io/github/dot166/jLib/internal/utils/LIBTest.java
package io.github.dot166.jLib.internal.utils;

import static io.github.dot166.jLib.utils.VersionUtils.getAndroidVersion;
import static io.github.dot166.jLib.jOS.Build.jOS_RELEASE;

import android.content.ComponentName;
import android.content.Intent;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.RestrictTo;
import androidx.appcompat.app.AppCompatActivity;

import io.github.dot166.jLib.internal.LIBTestBottomSheet;
import io.github.dot166.jLib.R;

@RestrictTo(RestrictTo.Scope.LIBRARY)
public class LIBTest {

    public static void Test(AppCompatActivity context) {
        if (context.getSupportActionBar() != null) {
            context.getSupportActionBar().setSubtitle("AAAA");
        }
        TextView text = context.findViewById(R.id.textView);
        Button button = context.findViewById(R.id.button);
        Button button2 = context.findViewById(R.id.button2);
        String android_lib_test = "Android: " + getAndroidVersion();
        String j_lib_test = "jOS: " + jOS_RELEASE;
        String all = android_lib_test + " " + j_lib_test;
        text.setText(all);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new LIBTestBottomSheet().show(context.getSupportFragmentManager(), LIBTestBottomSheet.TAG);
            }
        });
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                context.startActivity(new Intent().setComponent(new ComponentName(context.getApplicationInfo().packageName, "io.github.dot166.jLib.LIBChangelogActivity")));
            }
        });
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/internal/utils/LIBTest.kt
package io.github.dot166.jlib.internal.utils

import android.widget.Button
import android.widget.TextView
import androidx.annotation.RestrictTo
import androidx.appcompat.app.AppCompatActivity
import androidx.browser.customtabs.CustomTabsIntent
import androidx.core.net.toUri
import io.github.dot166.jlib.R
import io.github.dot166.jlib.internal.LIBTestBottomSheet
import io.github.dot166.jlib.jos.Build.jOS_RELEASE
import io.github.dot166.jlib.utils.VersionUtils.androidVersion
import io.github.dot166.jlib.utils.VersionUtils.libVersion

@RestrictTo(RestrictTo.Scope.LIBRARY)
object LIBTest {
    @JvmStatic
    fun Test(context: AppCompatActivity) {
        if (context.supportActionBar != null) {
            context.supportActionBar!!.subtitle = "flag_flag"
        }
        val text = context.findViewById<TextView>(R.id.textView)
        val button = context.findViewById<Button>(R.id.button)
        val button2 = context.findViewById<Button>(R.id.button2)
        val all = "Android: $androidVersion jOS: $jOS_RELEASE"
        text.text = all
        button.setOnClickListener {
            LIBTestBottomSheet().show(
                context.supportFragmentManager,
                LIBTestBottomSheet.TAG
            )
        }
        button2.setOnClickListener { v ->
            val webpage = ("https://github.com/dot166/jOS_j-lib/commits/$libVersion").toUri()
            val intent = CustomTabsIntent.Builder()
                .build()
            intent.launchUrl(v.context, webpage)
        }
    }
}
